### PR TITLE
B 2837 create matching questions flow

### PR DESCRIPTION
**Pull request by pixelfact**

### CODE REVIEW

File: app/project-recommendation/_features/matching-questions/matching-question.hooks.tsx
Comment (alexbeno) [inline app/project-recommendation/_features/matching-questions/matching-question.hooks.tsx:1]: Replace 

```
  currentQuestionIndex: number;
  selectedAnswers: Record<number, Answer[]>;
```

with string to don't convert UUID into number everywhere

File: app/project-recommendation/_features/matching-questions/matching-question.hooks.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      } else {
+        return {
+          ...prev,
+          selectedAnswers: {
+            ...prev.selectedAnswers,
+            [Number(currentQuestion.id)]: [{ ...answer, chosen: true }],
+          },
+        };
+      }
+    });
```
Comment (alexbeno) [inline app/project-recommendation/_features/matching-questions/matching-question.hooks.tsx:113]:
```suggestion
    setQuestionState(prev => {
      const currentAnswers = prev.selectedAnswers[Number(currentQuestion.id)] ?? [];

      if (currentQuestion.multipleChoice) {
        const existingAnswerIndex = currentAnswers.findIndex(a => a.index === answer.index);

        if (existingAnswerIndex >= 0) {
          return {
            ...prev,
            selectedAnswers: {
              ...prev.selectedAnswers,
              [Number(currentQuestion.id)]: currentAnswers.filter((_, i) => i !== existingAnswerIndex),
            },
          };
        } 
        
            return {
            ...prev,
            selectedAnswers: {
              ...prev.selectedAnswers,
              [Number(currentQuestion.id)]: [...currentAnswers, { ...answer, chosen: true }],
            },
          };
          
      }
      
         return {
          ...prev,
          selectedAnswers: {
            ...prev.selectedAnswers,
            [Number(currentQuestion.id)]: [{ ...answer, chosen: true }],
          },
        };
    });
```

File: core/domain/reco/reco-contract.types.ts [hunk 2]
```diff
@@ -21,3 +21,17 @@ export type GetMatchingQuestionsPortResponse = HttpStorageResponse<GetMatchingQu
...
+
+type SaveMatchingQuestionsPathParams = operations["saveMatchingQuestionAnswers"]["parameters"]["path"];
+type SaveMatchingQuestionsQueryParams = operations["saveMatchingQuestionAnswers"]["parameters"]["query"];
+
+export type SaveMatchingQuestionsPortParams = HttpClientParameters<{
+  PathParams: SaveMatchingQuestionsPathParams;
+  QueryParams: SaveMatchingQuestionsQueryParams;
+}>;
+
+export type SaveMatchingQuestionsPortResponse = HttpStorageResponse<never>;
```
Comment (haydencleary) [inline core/domain/reco/reco-contract.types.ts:37]:
```suggestion
export type SaveMatchingQuestionsPortResponse = HttpStorageResponse;
```

File: core/infrastructure/marketplace-api-client-adapter/adapters/reco-client-adapter.ts [hunk 3]
```diff
@@ -36,4 +37,24 @@ export class RecoClientAdapter implements RecoStoragePort {
...
+    const path = this.routes["saveMatchingQuestions"];
+    const method = "PUT";
+    const tag = HttpClient.buildTag({ path, pathParams, queryParams });
+
+    const request = async (body: SaveMatchingQuestionsBody) =>
+      this.client.request<never>({
+        path,
+        method,
+        tag,
+        pathParams,
```
Comment (haydencleary) [inline core/infrastructure/marketplace-api-client-adapter/adapters/reco-client-adapter.ts:51]: Missing queryParams

**Review by haydencleary — CHANGES_REQUESTED** [review]

**Review by haydencleary — APPROVED** [review]

**Review by alexbeno — APPROVED** [review]

**Pull request merged**